src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -219,29 +224,203 @@ export default class SourceFileLinter {
 			return;
 		}
 
-		if (ts.isPropertyDeclaration(rendererMember) && rendererMember.initializer) {
-			const initializerType = this.#checker.getTypeAtLocation(rendererMember.initializer);
+		if ((ts.isPropertyAssignment(rendererMember) || ts.isPropertyDeclaration(rendererMember) ||
+			ts.isVariableDeclaration(rendererMember)) && rendererMember.initializer) {
+			if (ts.isPropertyDeclaration(rendererMember)) {
+				const initializerType = this.#checker.getTypeAtLocation(rendererMember.initializer);
 
-			if (initializerType.flags & ts.TypeFlags.Undefined ||
-				initializerType.flags & ts.TypeFlags.Null) {
-				// null / undefined can be used to declare that a control does not have a renderer
-				return;
+				if (initializerType.flags & ts.TypeFlags.Undefined ||
+					initializerType.flags & ts.TypeFlags.Null) {
+					// null / undefined can be used to declare that a control does not have a renderer
+					return;
+				}
+
+				if (initializerType.flags & ts.TypeFlags.StringLiteral) {
+					let rendererName;
+					if (
+						ts.isStringLiteral(rendererMember.initializer) ||
+						ts.isNoSubstitutionTemplateLiteral(rendererMember.initializer)
+					) {
+						rendererName = rendererMember.initializer.text;
+					}
+					// Declaration as string requires sync loading of renderer module
+					this.#reporter.addMessage(MESSAGE.CONTROL_RENDERER_DECLARATION_STRING, {
+						className, rendererName,
+					}, rendererMember.initializer);
+				}
+			}
+
+			// Analyze renderer property when it's referenced by a variable or even another module
+			// i.e. { renderer: Renderer }
+			if (ts.isIdentifier(rendererMember.initializer)) {
+				const {symbol} = this.#checker.getTypeAtLocation(rendererMember);
+				const {declarations} = symbol ?? {};
+				declarations?.forEach((declaration) => this.analyzeControlRendererInternals(declaration));
+			} else {
+				// Analyze renderer property when it's directly embedded in the renderer object
+				// i.e. { renderer: {apiVersion: "2", render: () => {}} }
+				this.analyzeControlRendererInternals(rendererMember.initializer);
+			}
+		} else if (ts.isShorthandPropertyAssignment(rendererMember)) {
+			// Special case for shorthand property assignment. Basically the same as the Identifier case above
+			const {symbol: {declarations}} = this.#checker.getTypeAtLocation(rendererMember);
+			declarations?.forEach((declaration) => this.analyzeControlRendererInternals(declaration));
```

Comment:
This case is currently not covered by the tests

Reply:
This one is a bit hard to test now with the improved extend call transpilation, because of that line here:
https://github.com/SAP/ui5-linter/blob/31432b050a5b4a1dc446923331efa11b97652ef2/src/linter/ui5Types/amdTranspiler/rewriteExtendCall.ts#L67

I'll try to create a PR with a better handling of `shorthandPropertyAssignment`s

Reply:
https://github.com/SAP/ui5-linter/pull/385